### PR TITLE
Remove GetBlockSubsidy (Not Used)

### PR DESCRIPTION
**Pull request by KabDeveloper**
Removing GetBlockSubsidy (not used), already replaced by GetDogecoinBlockSubsidy !

Cleaning Code.

### CODE REVIEW

**Comment by KabDeveloper** [comment]
@rnicoll compiled dogecoin successfully with the new changes without any returned error.

Also it's logical, you can check that the method is not used nowhere.

**Comment by jimmyt1988** [comment]
I wonder if it's also worth removing these code comments referring to it. Just a thought: 

https://github.com/KabDeveloper/dogecoin/blob/3a5cc27dbd19eb01430e391f9ae07a5921d491eb/src/test/main_tests.cpp#L26

image

Or, just delete the commented out code altogether?

**Comment by KabDeveloper** [comment]
> I wonder if it's also worth removing these code comments referring to it. Just a thought:
> 
> https://github.com/KabDeveloper/dogecoin/blob/3a5cc27dbd19eb01430e391f9ae07a5921d491eb/src/test/main_tests.cpp#L26
> 
> image
> 
> Or, just delete the commented out code altogether?
> Perhaps even delete the method all together for that test and then all calling methods to that?

Thank you, I just removed the related/commented test methods from main_tests.cpp file.

File: src/test/main_tests.cpp [hunk 1]
```diff
@@ -13,54 +13,6 @@
...
-    //    int nHeight = nHalvings * consensusParams.nSubsidyHalvingInterval;
-    //    CAmount nSubsidy = GetBlockSubsidy(nHeight, consensusParams);
-    //    BOOST_CHECK(nSubsidy <= nInitialSubsidy);
-    //    BOOST_CHECK_EQUAL(nSubsidy, nPreviousSubsidy / 2);
-    //    nPreviousSubsidy = nSubsidy;
-    //}
-    //BOOST_CHECK_EQUAL(GetBlockSubsidy(maxHalvings * consensusParams.nSubsidyHalvingInterval, consensusParams), 0);
-}
-
-static void TestBlockSubsidyHalvings(int nSubsidyHalvingInterval)
```
Comment (jimmyt1988) [inline src/test/main_tests.cpp:34]: Looks good.

Variation in method signature has also been covered by the search (see `const Consensus::Params& consensusParams`):

image

File: src/test/main_tests.cpp [hunk 2]
```diff
@@ -13,54 +13,6 @@
...
-    //BOOST_CHECK_EQUAL(nPreviousSubsidy, nInitialSubsidy * 2);
-    //for (int nHalvings = 0; nHalvings < maxHalvings; nHalvings++) {
-    //    int nHeight = nHalvings * consensusParams.nSubsidyHalvingInterval;
-    //    CAmount nSubsidy = GetBlockSubsidy(nHeight, consensusParams);
-    //    BOOST_CHECK(nSubsidy <= nInitialSubsidy);
-    //    BOOST_CHECK_EQUAL(nSubsidy, nPreviousSubsidy / 2);
-    //    nPreviousSubsidy = nSubsidy;
-    //}
-    //BOOST_CHECK_EQUAL(GetBlockSubsidy(maxHalvings * consensusParams.nSubsidyHalvingInterval, consensusParams), 0);
-}
```
Comment (jimmyt1988) [inline src/test/main_tests.cpp:32]: Looks good:

Variation in method signature has also been covered by the search (see `int nSubsidyHalvingInterval`)

image

**Review by jimmyt1988 — APPROVED** [review]
Looks good to me! Nice refactoring! Hope we didn't miss something 😄

**Comment by rnicoll** [comment]
Re-triggering Travis

**Comment by rnicoll** [comment]
Sorry, still trying to get CI to pass. Changed the base branch and hoping that fixes it.

**Comment by KabDeveloper** [comment]
> Sorry, still trying to get CI to pass. Changed the base branch and hoping that fixes it.

Locally it's compiling without any error (Just like it was never modified).

On Ubuntu 18.04 / Boost-dev 1.68

**Comment by rnicoll** [comment]
Yeah, we had to tweak the Travis configuration, those changes are on a different branch, but now Travis isn't restarting the job now I've changed the base branch. It's vexing.

**Comment by rnicoll** [comment]
I've ended up rebasing it and raising a PR with the rebased commits, presuming that works I'll merge #2211 . It'll still credit you for the commits, it's just me managing the PR.

**Comment by KabDeveloper** [comment]
@rnicoll Ok, thank you !

**Comment by chromatic** [comment]
With #2211 merged, this can be closed and removed from "Needs Work" in the 1.14.4 project.

**Pull request closed without merging**